pkg/apis/core/v1/conversion.go

```diff
@@ -270,6 +274,117 @@ func Convert_v1_PodTemplateSpec_To_core_PodTemplateSpec(in *v1.PodTemplateSpec,
 	return nil
 }
 
+// helper for v1.PodIPInfo list to core.PodIPinfoList
+func v1_PodIPs_To_core_PodIPs(inList []v1.PodIPInfo) []core.PodIPInfo {
+	out := make([]core.PodIPInfo, len(inList))
+	for idx, value := range inList {
+		out[idx] = core.PodIPInfo{
+			IP:         value.IP,
+			Properties: value.Properties,
+		}
+	}
+	return out
+}
+
+// Converts podStatus from v1 to core
+// auto + handling podIp/podIps
+func Convert_v1_PodStatus_To_core_PodStatus(in *v1.PodStatus, out *core.PodStatus, s conversion.Scope) error {
+	if err := autoConvert_v1_PodStatus_To_core_PodStatus(in, out, s); err != nil {
+		return err
+	}
+
+	// handle ip/ips fields
+	if 0 < len(in.PodIP) && 0 < len(in.PodIPs) {
+		// both fields are provided
+		if in.PodIP != in.PodIPs[0].IP {
+			return fmt.Errorf("conversion Error: in.PodIP(%v) != in.PodIPs[0](%v)", in.PodIP, in.PodIPs[0].IP)
+		}
+		out.PodIPs = v1_PodIPs_To_core_PodIPs(in.PodIPs)
+	} else {
+		if 0 < len(in.PodIP) && 0 == len(in.PodIPs) {
+			// PodIP + NO entries in PodIPs
+			out.PodIPs = []core.PodIPInfo{
+				{
+					IP:         in.PodIP,
+					Properties: make(map[string]string),
+				},
+			}
+		} else {
+			// in.PodIP has no value
+			// copy the in.PodIPs
+			out.PodIPs = v1_PodIPs_To_core_PodIPs(in.PodIPs)
+		}
+	}
+
+	// Verify no duplicates in core.PodIPs
+	// We compare IPs not the entire struct
```

Comment:
if duplicate IPs are specified, isn't that an error we'd want to flag in validation, rather than silently dropping one of the records? I would have expected the data to be preserved and checked in validation, not dropped here

Reply:
I would agree to that. This was in the original spec, I wanted to have the discussion as we review. There are two things here: 
- The limits of bad behavior we would allow from CNI (such as duplicates) 
- conversion as per this comment https://github.com/kubernetes/enhancements/pull/808#discussion_r280239026

Both i am planning to address in review cycle.

btw thanks for this!

Reply:
Yeah, we should not be filtering it here.

Reply:
This has changed, validation moved to the right place..